packages/react-native-reanimated/Common/cpp/reanimated/NativeModules/ReanimatedModuleProxy.cpp

```diff
@@ -198,6 +198,9 @@ ReanimatedModuleProxy::~ReanimatedModuleProxy() {
   // runtime, so they have to go away before we tear down the runtime
   eventHandlerRegistry_.reset();
   frameCallbacks_.clear();
+#ifdef RCT_NEW_ARCH_ENABLED
+  operationsInBatch_.clear();
+#endif // RCT_NEW_ARCH_ENABLED
   uiWorkletRuntime_.reset();
```

Comment:
IMO this kind of pattern in the code - explicit cleanups - is going to resurface in the future. Can't we somehow slightly refactor this bit so we'd use RAII instead?

Reply:
I think we could move `uiWorkletRuntime_` in header file so that it is deallocated at the end, what do you think?